lib/os_places_api/client.rb

```diff
@@ -6,36 +6,30 @@ def initialize(token_manager)
       @token_manager = token_manager
     end
 
-    def locations_for_postcode(postcode, update: false)
+    def locations_for_postcode(postcode)
       postcode = PostcodeHelper.normalise(postcode)
 
-      if (record = Postcode.find_by(postcode: postcode)) && !update
+      if (record = Postcode.find_by(postcode: postcode))
         return build_locations(record["results"])
       end
 
-      response = HTTParty.get(
-        "https://api.os.uk/search/places/v1/postcode",
-        {
-          query: { postcode: postcode, output_srs: "WGS84" },
-          headers: { "Authorization": "Bearer #{@token_manager.access_token}" },
-        },
-      )
+      response = get_api_response(postcode)
+      raise UnexpectedResponse if response["results"].nil?
 
-      validate_response_code(response)
+      Postcode.create!(postcode: postcode, results: response["results"]) unless Postcode.find_by(postcode: postcode)
 
-      begin
-        json = JSON.parse(response.body)
-        raise UnexpectedResponse if json["results"].nil?
-
-        if (existing = Postcode.find_by(postcode: postcode))
-          existing.update(results: json["results"], updated_at: Time.now) if update
-        else
-          Postcode.create!(postcode: postcode, results: json["results"])
-        end
-        build_locations(json["results"])
-      rescue JSON::ParserError
-        raise UnexpectedResponse
-      end
+      build_locations(response["results"])
+    end
+
+    def update_postcode(postcode)
+      postcode = PostcodeHelper.normalise(postcode)
+
+      response = get_api_response(postcode)
+
+      record = Postcode.find_by(postcode: postcode)
+      raise UnexpectedResponse if response["results"].nil?
+
+      record.update(results: response["results"], updated_at: Time.now)
```

Comment:
We can lose the second parameter - that happens by default:

```suggestion
      record.update(results: response["results"])
```